File: 3726.remove-zeros-in-decimal-representation/solution.go
func removeZeros(n int64) int64 {
    res := int64(0)
    factor := int64(1)
    for ; n > 0; n /= 10 {
        if d := n % 10; d > 0 {
            res += d * factor
            factor *= 10
        }
    }
    return res
}

File: 3726.remove-zeros-in-decimal-representation/solution.kt
class Solution {
    fun removeZeros(n: Long): Long {
        var rest = n
        var factor = 1L
        var res = 0L

        while (rest > 0) {
            val digit = rest % 10
            if (digit > 0) {
                res += digit * factor
                factor *= 10
            }
            rest /= 10
        }

        return res
    }
}